drivers/soundwire/mipi_disco.c

```diff
@@ -23,6 +23,26 @@
 #include <linux/soundwire/sdw.h>
 #include "bus.h"
 
+static bool mipi_fwnode_property_read_bool(const struct fwnode_handle *fwnode,
+					   const char *propname)
+{
+	int ret;
+	u8 val;
+
+	if (!fwnode_property_present(fwnode, propname))
+		return false;
+	ret = fwnode_property_read_u8_array(fwnode, propname, &val, 1);
```

Comment:
Hmm… we should be careful with endianness here. DT keeps values in BE32 format, and if this code ever be run on OF based system, it might be problematic. But despite that, it’s always better to be in a robust side. Up to you , though, how to improve this, if necessary at all.

Reply:
so far all MIPI DisCo definitions rely on ACPI, there's no support for DeviceTree at all. The main reason is that no one on the committee knew anything about DeviceTree, all we had was ACPI tools. Maybe in a future revision this will change, but that depends on participation in a standard so will take time.

Reply:
I will add a note though to mention the endianess and make sure this is documented.

That's also another reason why it makes sense to keep those helpers MIPI/DisCo specific, I am not sure how we could change the common helper.

Reply:
commit message modified to add note on endianness, thanks @andy-shev for the feeedback.